pkg/playbook/app/installer/crypto.go

```diff
@@ -93,16 +93,33 @@ func prepareCertificateForBundle(request certificate.Request, pcc certificate.PE
 		zap.L().Debug("CSR Origin is [local]. Private Key added to PEM Collection")
 	}
 
-	//Key needs to be decrypted in order to create the bundle (PKCS12, JKS)
+	// Key needs to be decrypted in order to create the bundle (PKCS12, JKS)
 	// Firefly does not encrypt Private Keys. Thus, Private Key should not be decrypted in that scenario
 	if pcc.PrivateKey != "" && decryptPK {
+
 		privateKey, err := vcertutil.DecryptPrivateKey(pcc.PrivateKey, request.KeyPassword)
 		if err != nil {
-			return nil, err
+			// there's chance we've got a private key that is not PKCS8
+			zap.L().Debug("PKCS8 decrypt failed for opening Private Key. Trying legacy decrypt")
+			privKey, err := getPrivateKey(pcc.PrivateKey, request.KeyPassword)
+			if err != nil {
+				return nil, err
+			}
+
+			var privKeyBlock *pem.Block
+
+			privKeyD := privKey.(crypto.Signer)
+			privKeyBlock, err = certificate.GetPrivateKeyPEMBock(privKeyD, util.LegacyPem)
+			if err != nil {
+				return nil, err
+			}
+			privEncodedPEM := pem.EncodeToMemory(privKeyBlock)
+			privKeyPEMString := string(privEncodedPEM)
+			pcc.PrivateKey = privKeyPEMString
+		} else {
```

Comment:
You can remove the `else` clause, assign `privKeyPEMString` to privateKey.
on line 118:
privateKey = privKeyPEMString
then remove line 119 and 121.

Reply:
Done by latest commit